src/main/kotlin/org/wfanet/measurement/system/v1alpha/ComputationKey.kt

```diff
@@ -0,0 +1,36 @@
+// Copyright 2021 The Cross-Media Measurement Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.wfanet.measurement.system.v1alpha
+
+import org.wfanet.measurement.common.ResourceNameParser
+
+private val parser = ResourceNameParser("computations/{computation}")
+
+/** [ResourceKey] of a Computation. */
+data class ComputationKey(val computationId: String) : ResourceKey {
+  override fun toName(): String {
+    return parser.assembleName(mapOf(IdVariable.COMPUTATION to computationId))
+  }
+
+  companion object {
+    val defaultValue = ComputationKey("")
```

Comment:
Since protobuf name fields don't have a distinction between unset and default value (empty string), we need a way to represent that as a ResourceKey. This basically emulates the behavior of a protobuf message.

They're used in a child PR that switches the system API to have resource names rather than keys. In that PR, there are `key` extension properties that allow some of the existing code to function mostly unchanged (aside from cases where there are incorrect checks of whether a protobuf string field is null instead of empty, as protobuf string fields can't be null).

You're right that toName() should throw an error in this case. I'll handle that in ResoureNameParser.